public/src/hooks.js

```diff
@@ -1,12 +1,16 @@
 import * as axios from 'axios';
 
 const axiosConfig = {
-  baseURL: document.getElementById('server-URL').value,
+  // eslint-disable-next-line no-undef
+  baseURL: SERVER_CONFIGURATION.baseUrl,
 };
 export const API = {
   axiosInstance: axios.create(axiosConfig),
+  // eslint-disable-next-line no-undef
+  configureUrl: (url) => `${SERVER_CONFIGURATION.baseUrl}${url}`,
+  post: (url, data) => fetch(url, data),
   submit(data) {
-    return this.axiosInstance.post('records/', data);
+    return this.post(this.configureUrl('records/'), data);
```

Comment:
are you forgetting to `.json`? :)

Reply:
i was debating that, because unless I'm mistaken, calling `.json` when it network fails would throw an error yeah?

So I figured I'd return the promise untouched, and then the consumer can try to do the `.json` themselves.

Reply:
Nope, this is a promise, so you would have

const response = await fetch(endpoint)
return response.json()

if the network fails, it'll throw a runtime exception and not invoke .json, but since this would be an async function, it would just return a rejected promise (just as if you hadn't invoked .json)

Reply:
gotcha. Changing it now.

Reply:
Done.